independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -403,8 +413,26 @@ public Builder addRemovalExclusion(Predicate<BeanInfo> exclusion) {
             return this;
         }
 
+        /**
+         * 
+         * @param removeFinalForProxyableMethods
+         * @return self
+         * @deprecated This method will be removed at some point post Quarkus 1.4
+         */
+        @Deprecated
         public Builder setRemoveFinalFromProxyableMethods(boolean removeFinalForProxyableMethods) {
-            this.removeFinalForProxyableMethods = removeFinalForProxyableMethods;
+            return this;
+        }
+
+        /**
+         * If set to true the container will attempt to remove the final flag from a bean class that must be proxyable during
```

Comment:
I think this comment needs to be augmented right? Since the transformation isn't only about removing final anymore